Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.h

```diff
@@ -170,11 +176,31 @@ class NumberFormat final : public Object {
     Notation m_notation { Notation::Invalid };              // [[Notation]]
     Optional<CompactDisplay> m_compact_display {};          // [[CompactDisplay]]
     SignDisplay m_sign_display { SignDisplay::Invalid };    // [[SignDisplay]]
+    NativeFunction* m_bound_format { nullptr };             // [[BoundFormat]]
+};
+
+struct FormatResult {
+    String formatted_string;       // [[FormattedString]]
+    double rounded_number { 0.0 }; // [[RoundedNumber]]
+};
+
+struct RawFormatResult : public FormatResult {
+    int digits { 0 }; // [[IntegerDigitsCount]]
```

Comment:
does this need to be signed?

Reply:
I *think* I made these signed because they are compared against numbers from the wild which are also stored as signed. *But* those numbers are later capped between something like 1 and 21, so they can't be unsigned either. So yeah, that can all be made unsigned.

Reply:
Oh, actually, these values are used in signed arithmetic. The values themselves can't be negative, but some math that is done with them can have negative results.